%% bare_conf.tex
%% V1.4a
%% 2014/09/17
%% by Michael Shell
%% See:
%% http://www.michaelshell.org/
%% for current contact information.
%%
%% This is a skeleton file demonstrating the use of IEEEtran.cls
%% (requires IEEEtran.cls version 1.8a or later) with an IEEE
%% conference paper.
%%
%% Support sites:
%% http://www.michaelshell.org/tex/ieeetran/
%% http://www.ctan.org/tex-archive/macros/latex/contrib/IEEEtran/
%% and
%% http://www.ieee.org/

%%*************************************************************************
%% Legal Notice:
%% This code is offered as-is without any warranty either expressed or
%% implied; without even the implied warranty of MERCHANTABILITY or
%% FITNESS FOR A PARTICULAR PURPOSE!
%% User assumes all risk.
%% In no event shall IEEE or any contributor to this code be liable for
%% any damages or losses, including, but not limited to, incidental,
%% consequential, or any other damages, resulting from the use or misuse
%% of any information contained here.
%%
%% All comments are the opinions of their respective authors and are not
%% necessarily endorsed by the IEEE.
%%
%% This work is distributed under the LaTeX Project Public License (LPPL)
%% ( http://www.latex-project.org/ ) version 1.3, and may be freely used,
%% distributed and modified. A copy of the LPPL, version 1.3, is included
%% in the base LaTeX documentation of all distributions of LaTeX released
%% 2003/12/01 or later.
%% Retain all contribution notices and credits.
%% ** Modified files should be clearly indicated as such, including  **
%% ** renaming them and changing author support contact information. **
%%
%% File list of work: IEEEtran.cls, IEEEtran_HOWTO.pdf, bare_adv.tex,
%%                    bare_conf.tex, bare_jrnl.tex, bare_conf_compsoc.tex,
%%                    bare_jrnl_compsoc.tex, bare_jrnl_transmag.tex
%%*************************************************************************


% *** Authors should verify (and, if needed, correct) their LaTeX system  ***
% *** with the testflow diagnostic prior to trusting their LaTeX platform ***
% *** with production work. IEEE's font choices and paper sizes can       ***
% *** trigger bugs that do not appear when using other class files.       ***                          ***
% The testflow support page is at:
% http://www.michaelshell.org/tex/testflow/



\documentclass[conference,a4paper]{IEEEtran}
% Some Computer Society conferences also require the compsoc mode option,
% but others use the standard conference format.
%
% If IEEEtran.cls has not been installed into the LaTeX system files,
% manually specify the path to it like:
% \documentclass[conference]{../sty/IEEEtran}





% Some very useful LaTeX packages include:
% (uncomment the ones you want to load)


% *** MISC UTILITY PACKAGES ***
%
%\usepackage{ifpdf}
% Heiko Oberdiek's ifpdf.sty is very useful if you need conditional
% compilation based on whether the output is pdf or dvi.
% usage:
% \ifpdf
%   % pdf code
% \else
%   % dvi code
% \fi
% The latest version of ifpdf.sty can be obtained from:
% http://www.ctan.org/tex-archive/macros/latex/contrib/oberdiek/
% Also, note that IEEEtran.cls V1.7 and later provides a builtin
% \ifCLASSINFOpdf conditional that works the same way.
% When switching from latex to pdflatex and vice-versa, the compiler may
% have to be run twice to clear warning/error messages.






% *** CITATION PACKAGES ***
%
\usepackage{cite}
% cite.sty was written by Donald Arseneau
% V1.6 and later of IEEEtran pre-defines the format of the cite.sty package
% \cite{} output to follow that of IEEE. Loading the cite package will
% result in citation numbers being automatically sorted and properly
% "compressed/ranged". e.g., [1], [9], [2], [7], [5], [6] without using
% cite.sty will become [1], [2], [5]--[7], [9] using cite.sty. cite.sty's
% \cite will automatically add leading space, if needed. Use cite.sty's
% noadjust option (cite.sty V3.8 and later) if you want to turn this off
% such as if a citation ever needs to be enclosed in parenthesis.
% cite.sty is already installed on most LaTeX systems. Be sure and use
% version 5.0 (2009-03-20) and later if using hyperref.sty.
% The latest version can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/contrib/cite/
% The documentation is contained in the cite.sty file itself.

\usepackage{hyperref}
\hypersetup{
    % bookmarks=true,         % show bookmarks bar?
    unicode=true,          % non-Latin characters in Acrobat’s bookmarks
    pdftoolbar=true,        % show Acrobat’s toolbar?
    pdfmenubar=true,        % show Acrobat’s menu?
    pdffitwindow=false,     % window fit to page when opened
    pdfstartview={FitH},    % fits the width of the page to the window
    %pdftitle={},    % title
    %pdfauthor={Author},     % author
    %pdfsubject={Subject},   % subject of the document
    %pdfcreator={Creator},   % creator of the document
    %pdfproducer={Producer}, % producer of the document
    %pdfkeywords={keyword1, key2, key3}, % list of keywords
    %pdfnewwindow=true,      % links in new PDF window
    colorlinks=true,       % false: boxed links; true: colored links
    linkcolor=black,          % color of internal links (change box color with linkbordercolor)
    citecolor=black,        % color of links to bibliography
    filecolor=black,      % color of file links
    urlcolor=black,           % color of external links
    final=true
  }




% *** GRAPHICS RELATED PACKAGES ***
%
\ifCLASSINFOpdf
  % \usepackage[pdftex]{graphicx}
  % declare the path(s) where your graphic files are
  % \graphicspath{{../pdf/}{../jpeg/}}
  % and their extensions so you won't have to specify these with
  % every instance of \includegraphics
  % \DeclareGraphicsExtensions{.pdf,.jpeg,.png}
\else
  % or other class option (dvipsone, dvipdf, if not using dvips). graphicx
  % will default to the driver specified in the system graphics.cfg if no
  % driver is specified.
  % \usepackage[dvips]{graphicx}
  % declare the path(s) where your graphic files are
  % \graphicspath{{../eps/}}
  % and their extensions so you won't have to specify these with
  % every instance of \includegraphics
  % \DeclareGraphicsExtensions{.eps}
\fi
% graphicx was written by David Carlisle and Sebastian Rahtz. It is
% required if you want graphics, photos, etc. graphicx.sty is already
% installed on most LaTeX systems. The latest version and documentation
% can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/required/graphics/
% Another good source of documentation is "Using Imported Graphics in
% LaTeX2e" by Keith Reckdahl which can be found at:
% http://www.ctan.org/tex-archive/info/epslatex/
%
% latex, and pdflatex in dvi mode, support graphics in encapsulated
% postscript (.eps) format. pdflatex in pdf mode supports graphics
% in .pdf, .jpeg, .png and .mps (metapost) formats. Users should ensure
% that all non-photo figures use a vector format (.eps, .pdf, .mps) and
% not a bitmapped formats (.jpeg, .png). IEEE frowns on bitmapped formats
% which can result in "jaggedy"/blurry rendering of lines and letters as
% well as large increases in file sizes.
%
% You can find documentation about the pdfTeX application at:
% http://www.tug.org/applications/pdftex





% *** MATH PACKAGES ***
%
%\usepackage[cmex10]{amsmath}
% A popular package from the American Mathematical Society that provides
% many useful and powerful commands for dealing with mathematics. If using
% it, be sure to load this package with the cmex10 option to ensure that
% only type 1 fonts will utilized at all point sizes. Without this option,
% it is possible that some math symbols, particularly those within
% footnotes, will be rendered in bitmap form which will result in a
% document that can not be IEEE Xplore compliant!
%
% Also, note that the amsmath package sets \interdisplaylinepenalty to 10000
% thus preventing page breaks from occurring within multiline equations. Use:
%\interdisplaylinepenalty=2500
% after loading amsmath to restore such page breaks as IEEEtran.cls normally
% does. amsmath.sty is already installed on most LaTeX systems. The latest
% version and documentation can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/required/amslatex/math/





% *** SPECIALIZED LIST PACKAGES ***
%
%\usepackage{algorithmic}
% algorithmic.sty was written by Peter Williams and Rogerio Brito.
% This package provides an algorithmic environment fo describing algorithms.
% You can use the algorithmic environment in-text or within a figure
% environment to provide for a floating algorithm. Do NOT use the algorithm
% floating environment provided by algorithm.sty (by the same authors) or
% algorithm2e.sty (by Christophe Fiorio) as IEEE does not use dedicated
% algorithm float types and packages that provide these will not provide
% correct IEEE style captions. The latest version and documentation of
% algorithmic.sty can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/contrib/algorithms/
% There is also a support site at:
% http://algorithms.berlios.de/index.html
% Also of interest may be the (relatively newer and more customizable)
% algorithmicx.sty package by Szasz Janos:
% http://www.ctan.org/tex-archive/macros/latex/contrib/algorithmicx/




% *** ALIGNMENT PACKAGES ***
%
%\usepackage{array}
% Frank Mittelbach's and David Carlisle's array.sty patches and improves
% the standard LaTeX2e array and tabular environments to provide better
% appearance and additional user controls. As the default LaTeX2e table
% generation code is lacking to the point of almost being broken with
% respect to the quality of the end results, all users are strongly
% advised to use an enhanced (at the very least that provided by array.sty)
% set of table tools. array.sty is already installed on most systems. The
% latest version and documentation can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/required/tools/


% IEEEtran contains the IEEEeqnarray family of commands that can be used to
% generate multiline equations as well as matrices, tables, etc., of high
% quality.




% *** SUBFIGURE PACKAGES ***
%\ifCLASSOPTIONcompsoc
%  \usepackage[caption=false,font=normalsize,labelfont=sf,textfont=sf]{subfig}
%\else
%  \usepackage[caption=false,font=footnotesize]{subfig}
%\fi
% subfig.sty, written by Steven Douglas Cochran, is the modern replacement
% for subfigure.sty, the latter of which is no longer maintained and is
% incompatible with some LaTeX packages including fixltx2e. However,
% subfig.sty requires and automatically loads Axel Sommerfeldt's caption.sty
% which will override IEEEtran.cls' handling of captions and this will result
% in non-IEEE style figure/table captions. To prevent this problem, be sure
% and invoke subfig.sty's "caption=false" package option (available since
% subfig.sty version 1.3, 2005/06/28) as this is will preserve IEEEtran.cls
% handling of captions.
% Note that the Computer Society format requires a larger sans serif font
% than the serif footnote size font used in traditional IEEE formatting
% and thus the need to invoke different subfig.sty package options depending
% on whether compsoc mode has been enabled.
%
% The latest version and documentation of subfig.sty can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/contrib/subfig/




% *** FLOAT PACKAGES ***
%
%\usepackage{fixltx2e}
% fixltx2e, the successor to the earlier fix2col.sty, was written by
% Frank Mittelbach and David Carlisle. This package corrects a few problems
% in the LaTeX2e kernel, the most notable of which is that in current
% LaTeX2e releases, the ordering of single and double column floats is not
% guaranteed to be preserved. Thus, an unpatched LaTeX2e can allow a
% single column figure to be placed prior to an earlier double column
% figure. The latest version and documentation can be found at:
% http://www.ctan.org/tex-archive/macros/latex/base/


%\usepackage{stfloats}
% stfloats.sty was written by Sigitas Tolusis. This package gives LaTeX2e
% the ability to do double column floats at the bottom of the page as well
% as the top. (e.g., "\begin{figure*}[!b]" is not normally possible in
% LaTeX2e). It also provides a command:
%\fnbelowfloat
% to enable the placement of footnotes below bottom floats (the standard
% LaTeX2e kernel puts them above bottom floats). This is an invasive package
% which rewrites many portions of the LaTeX2e float routines. It may not work
% with other packages that modify the LaTeX2e float routines. The latest
% version and documentation can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/contrib/sttools/
% Do not use the stfloats baselinefloat ability as IEEE does not allow
% \baselineskip to stretch. Authors submitting work to the IEEE should note
% that IEEE rarely uses double column equations and that authors should try
% to avoid such use. Do not be tempted to use the cuted.sty or midfloat.sty
% packages (also by Sigitas Tolusis) as IEEE does not format its papers in
% such ways.
% Do not attempt to use stfloats with fixltx2e as they are incompatible.
% Instead, use Morten Hogholm'a dblfloatfix which combines the features
% of both fixltx2e and stfloats:
%
% \usepackage{dblfloatfix}
% The latest version can be found at:
% http://www.ctan.org/tex-archive/macros/latex/contrib/dblfloatfix/




% *** PDF, URL AND HYPERLINK PACKAGES ***
%
%\usepackage{url}
% url.sty was written by Donald Arseneau. It provides better support for
% handling and breaking URLs. url.sty is already installed on most LaTeX
% systems. The latest version and documentation can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/contrib/url/
% Basically, \url{my_url_here}.




% *** Do not adjust lengths that control margins, column widths, etc. ***
% *** Do not use packages that alter fonts (such as pslatex).         ***
% There should be no need to do such things with IEEEtran.cls V1.6 and later.
% (Unless specifically asked to do so by the journal or conference you plan
% to submit to, of course. )


% correct bad hyphenation here
\hyphenation{op-tical net-works semi-conduc-tor}

\providecommand\url[1]{\texttt{#1}}

\begin{document}
\urlstyle{tt}

%
% paper title
% Titles are generally capitalized except for words such as a, an, and, as,
% at, but, by, for, in, nor, of, on, or, the, to and up, which are usually
% not capitalized unless they are the first or last word of the title.
% Linebreaks \\ can be used within to get better formatting as desired.
% Do not put math or special symbols in the title.
\date{}
\title{Toward Big Data Analysis in the Baikal Microbiome Study}

%\author{Evgeny Cherkashin, Alexey Kopaigorodsky, Ljubica Kazi.}

% author names and affiliations
% use a multiple column layout for up to three different
% affiliations
% \author{\IEEEauthorblockN{Michael Shell}
% \IEEEauthorblockA{School of Electrical and\\Computer Engineering\\
% Georgia Institute of Technology\\
% Atlanta, Georgia 30332--0250\\
% Email: http://www.michaelshell.org/contact.html}
% \and
% \IEEEauthorblockN{Homer Simpson}
% \IEEEauthorblockA{Twentieth Century Fox\\
% Springfield, USA\\
% Email: homer@thesimpsons.com}
% \and
% \IEEEauthorblockN{James Kirk\\ and Montgomery Scott}
% \IEEEauthorblockA{Starfleet Academy\\
% San Francisco, California 96678--2391\\
% Telephone: (800) 555--1212\\
% Fax: (888) 555--1212}}

% conference papers do not typically use \thanks and this command
% is locked out in conference mode. If really needed, such as for
% the acknowledgment of grants, issue a \IEEEoverridecommandlockouts
% after \documentclass

% for over three affiliations, or if they all won't fit within the width
% of the page, use this alternative format:
%

\DeclareRobustCommand*{\IEEEauthorrefmark}[1]{\raisebox{0pt}[0pt][0pt]{\textsuperscript{\footnotesize #1}}}

\author{\IEEEauthorblockN{Fedor Malkov\IEEEauthorrefmark{1,}\IEEEauthorrefmark{2},
Evgeny Cherkashin\IEEEauthorrefmark{1,}\IEEEauthorrefmark{2,}\IEEEauthorrefmark{3,}\IEEEauthorrefmark{5},
Sergey Gorsky\IEEEauthorrefmark{2,}\IEEEauthorrefmark{3},
Alexey Shigarov\IEEEauthorrefmark{1,}\IEEEauthorrefmark{3},\\
Yury Galachyants\IEEEauthorrefmark{1,}\IEEEauthorrefmark{4},
Alexey Morozov\IEEEauthorrefmark{1,}\IEEEauthorrefmark{4},
Yelena Likhoshway\IEEEauthorrefmark{1,}\IEEEauthorrefmark{4},
and
Ivan Mikhailov\IEEEauthorrefmark{1,}\IEEEauthorrefmark{4}}\\[-0.9em]
%
\IEEEauthorblockA{\IEEEauthorrefmark{1}Irkutsk Scientific Center of SB RAS, 134 Lermontov Street, Russia, 664033}
\IEEEauthorblockA{\IEEEauthorrefmark{2}Matrosov Institute for System Dynamics and Control Theory of SB RAS, 134 Lermontov Street, Russia, 664033}
\IEEEauthorblockA{\IEEEauthorrefmark{3}Irkutsk State University, 20 Gagarina Avenue, Russia, 664002}
\IEEEauthorblockA{\IEEEauthorrefmark{4}Limnological Institute of SB RAS, 3 Ulan-Bator Street, Russia, 664033}
\IEEEauthorblockA{\IEEEauthorrefmark{5}National Research Irkutsk State Technical University, 83 Lermontov Street, Russia, 664074}
E-mail:\texttt{\{malkov,eugeneai,gorsky,shig\}@icc.ru,\{morozov,mikhailov\}@lin.irk.ru}}

% use for special paper notices
%\IEEEspecialpapernotice{(Invited Paper)}




% make the title area
\maketitle

% As a general rule, do not put math, special symbols or citations
% in the abstract
\begin{abstract}
  The Baikal microbiome study is being conducted in Limnological Institute of SB RAS, the study is based on the next-generation sequencing data analysis. As a result, the study generates a big volume of metagenomic data. A seasonal monitoring of Baikal microbiome requires the study of thousands of samples. The process of metagenomic data analysis is multi-stage, e.g. for amplicons up to 50 consecutive steps can be included. This requires microbiologist to be able to construct and execute a pipeline of the processing, which consists of command-line utilities like MOTHUR or QIIME, scripts of the Python programming language, statistical processing with R modules. Additionally, the execution of the pipeline requires interaction with a storage and management software of metagenomic data. At the current state of the automation of the analysis, microbiologist is not always able to cope with the complex task independently. In this work, we present an environment that supports visual planning and parallel execution of pipelines of the metagenomic data analysis. The visual planning is implemented as a plug-in module of RAPID MINER system, and the parallel execution is provided by the instrumental system ORLANDO. The big data storage facility is based on Hadoop infrastructure. The environment under construction simplifies the research activity of microbiologist, making the metagenomic analysis to be more accessible for researcher without in-depth programming and data science skills.
\end{abstract}

%%%% KEYWORDS spreadsheet data transformation, table analysis, table interpretation, rule-based systems

% no keywords




% For peer review papers, you can put extra information on the cover
% page as needed:
% \ifCLASSOPTIONpeerreview
% \begin{center} \bfseries EDICS Category: 3-BBND \end{center}
% \fi
%
% For peerreview papers, this IEEEtran command inserts a page break and
% creates the second title. It will be ignored for other modes.
\IEEEpeerreviewmaketitle



\section{Introduction}
% no \IEEEPARstart

\paragraph{FROM ABSTRACT}

The Baikal microbiome study is being conducted in Limnological Institute of SB RAS, the study is based on the next-generation sequencing data analysis. As a result, the study generates a big volume of metagenomic data. A seasonal monitoring of Baikal microbiome requires the study of thousands of samples. The process of metagenomic data analysis is multi-stage, e.g. for amplicons up to 50 consecutive steps can be included. This requires microbiologist to be able to construct and execute a pipeline of the processing, which consists of command-line utilities like MOTHUR or QIIME, scripts of the Python programming language, statistical processing with R modules. Additionally, the execution of the pipeline requires interaction with a storage and management software of metagenomic data. At the current state of the automation of the analysis, microbiologist is not always able to cope with the complex task independently. In this work, we present an environment that supports visual planning and parallel execution of pipelines of the metagenomic data analysis. The visual planning is implemented as a plug-in module of RAPID MINER system, and the parallel execution is provided by the instrumental system ORLANDO. The big data storage facility is based on Hadoop infrastructure. The environment under construction simplifies the research activity of microbiologist, making the metagenomic analysis to be more accessible for researcher without in-depth programming and data science skills.


In the last decade, thanks to the invention of next-generation sequencing (NGS) methods and their introduction in practice of research of biological systems a field of research of molecular genetics, namely metagenomics, has been arisen. Its basic principle is that the object under investigation is not a separate microscopic organism, but their communities (microbiomes). The sampled probes stand out with a total DNA sequencing data over the whole set of genes of all microorganisms in the probe. That is, the studied object is the microbiome as a whole, not only those organisms which can be cultivated in laboratory conditions or identified with microscopic or microbiological methods.

Metagenomics allows us to describe a significant number of new groups on all taxonomic levels, broadening the field of view of the world science. A characteristic example is the recently discovered group CPR (candidate phyla radiation). No CPR is isolated in a culture at the moment. According to genomic data, its representatives differ in the set of ribosomal proteins, the absence of certain key metabolic pathways and the presence of self-splicing introns in genes 16S rRNA [1]. Phylogenetic analysis indicates that this group is a sister to all other bacteria, and the level of divergence is not inferior to bacteria, not to mention the eukaryotes [2].

There are two main types of metagenomic studies. The first one, which is simpler, is called analysis of the amplicons. In this case a specific taxonomic marker is amplified and sequenced. The marker is universal for the studied species. Usually, the sequence of the small subunit of ribosomal RNA is used as the marker, as this gene is widely used in phylogenetics. The gene is available in numerous reference sequences. For example, the release 128 of widely used in amplicon analysis SILVA database [3] contains 645 151 unique rRNA. The reads obtained from the DNA sequences extracted from the sample under investigation are compared to the sequences in databases, attributing them to a particular taxon of a taxonomic level, obtaining information about the diversity of the microbiome in the studied environment.

The second approach is known as metogenomic shotgun method. It is based on sequencing the whole DNA sample instead of the specific locus. With sufficient coverage, this approach allows describing the taxonomic composition of the community, as well as the genes of functional or structural proteins presented in the representatives of the community, including viral ones [4]. On the basis of metagenomic data, metabolic interactions in individual microbiomes can be determined using the databases ePGDBs (environmental pathway/genome databases) [5]. In several works, full genomes of individual species were isolated from metagenomic dataset reads [6].
In recent years the amplicon analysis was applied in microbiome studies for different environments of lake Baikal. The researchers of the Limnological Institute of SB RAS described the under-ice bacterial communities associated with blooms of diatoms [7] and bacteria in photic layer during spring [8]. Bacteria inhabiting the Baikal sponges were studied as well [9]. Finally, the bacterial communities of bottom sediments in the areas of hydrocarbon yields [10,11] were investigated.

In order to carry out the metagenomic studies the significant computational resources and bioinformatics skills are required for data processing and interpretation. The software used for analysis of amplicons includes various library modules of sequence processing, for example, Mothur [12], USearch [13], statistical packages and development environments of data mining algorithms, e.g., R (https://www.r-project.org). In order to carry out the studies of metagenomic data, the specialists are required to be able scripting the command shell of an operating system (Linux, Windows), running programs in a distributed computing environment and cluster computing systems, and programming with general-purpose languages, usually R or Python.

Another important problem is the organization of a centralized data storage and providing the efficient regulated access to the data for the users.  At the moment the staff of LIN SB RAS conducted numerous amplicon research of the different ecotopes of lake Baikal, the data were collected for several years. There are no strict rules of the storage policy of input, intermediate data and the obtained results. Comparison and integration of data from different studies is also complicated due to its heterogeneity, resulting from the use of various software. The implementation of a system for storing input data, metadata, and results of metagenomic studies in a unified form will simplify the integration of results from different studies and the comparative analysis.
The goal of this study is a software environment development for supporting the processes of new-generation sequencing with organizational, informational and computational resources.

\section{The domain analysis}
\label{sec:doma}


Domain analysis showed that the problems solved in the bioinformational part of metagenomic analysis, together with NGS itself, are well represented within the paradigm of Big Data. At the moment, the scientific community developed data formats for representation and storage of metagenomic information, algorithms and software modules including distributed and parallel implementations on cluster computing systems providing different stages of data analysis.

The solution of the problems within the Big Data paradigm requires the biologist to have software development skills to be a professional programmer in bioinformatics. In order to carry on the analysis of each probe, biologist is to construct and execute a separate program script or perform stage-by-stage execution manually to control each step's results quality. This approach significantly slows down the process of obtaining the final result.

The proposed organization of studies is based on the creation of an information-computational environment that allows one to design and execute scenarios, giving the input data in various formats from various sources, e.g., files, databases, servers of metagenomic information. The environment must also support a cloud storage for intermediate data and the obtained results. A collaborative project of LIN SB RAS and ISDCT SB RAS is devoted to the construction of the environment for the research support. The following problems are to be solved within the project.
\begin{enumerate}
\item The subject area and its functional modeling. The classes of
  functions (problems) are being recognized and presented in the form
  of software modules. Modules form scripts of problem solving,
  network graphs of modules connected by data transmission.
\item Metadata descriptions of the modules and structures of input and
  output data. At this stage, it is necessary to deal with the problem
  of integration with external information and computational
  resources. In this case, the standards and standard means of data
  modeling like ontologies are of critical usage.
\item Decomposition of
  the input/output data formats and implementation of subsystems of
  their transformation, accumulation, storage and effective (according
  to the criteria of time and computational complexity) regulated
  access.
\item Construction of virtual executional environments and
  software interfaces for modules, whose source code is inaccessible
  due to the lack of the source code or licensing restrictions.
\item Development a customized user interface for high level control of
  the scenario executions. At this stage, a visual programming with
  the user interface for script development and execution is required
  to provide flexibility for managing computational processes by
  domain specialists.
\item Development of subsystems of visualization
  and interpretation of obtained results, including the modules for
  interpretation of the process of metagenomic analysis.
\item Dataflow
  representation of the domain A popular approach to the
  representation of the computational process is dataflow programming
  [14]. The data flow programs are constructed as a combination of the
  executable modules. The modules receive input data, process it, and
  produce output. The approach is being developed since the 1970-ies.
\end{enumerate}
An example of usage of the script construction system under development is shown in Fig. 1. The figure shows an example of an initial stage of a computing process of analysis of the amplicons.

\begin{figure}[t]\centering

Fig. 1.  \caption{The initial stage of the metagenomic analysis represented as a data flow.}
\end{figure}

The presented script was constructed by means of the software package Rapidminer Studio (https://rapidminer.com/) supplemented by our extension module for description of the amplicon analysis stages. The scenario includes the following operations:
\begin{itemize}
\item definition of a research project as a set of input files
  containing the sequencing data in a directory (module “Input”);
\item trimming reads (module “Trim”);
\item the module “Summary” is used for
  visual quality analysis of the results of the previous steps;
\item the
  reduction of the volume of input by the removal of insignificant
  information, for example, overlapping sequences (modules,
  “Uniq. select …”);
\item alignment of sequences to the reference
  database (module “Alignment”);
\item filtering sequences according to
  specified criteria (module “Screening”);
\item removing alignment
  columns based on specified criteria, for example, empty columns
  (module “Filter”);
\item removal of sequences containing sequencing
  error (the module “Scrap”);
\item detection of chimeras (module ``Chim. detection''), \emph{etc}.
  \end{itemize}
  The diagram shows service modules of RapidMiner Studio, which are necessary to distribute the same type of information between modules (e.g., “Copy groups”). The necessity of introduction of such modules is a feature of Rapidminer Studio; it supposes that in a general case the modules make changes in the data under processing without copying it.

Each module receives file names as input and creates new file set as the result. The operation of the module depends on the parameters specified by the user via user interface of each module. The results of the script are sent to output ports and displayed by the Rapidminer Studio visualization subsystem in a convenient form to the user. The system supports presentation of a scenario as a new block with its input and output ports, as well as a cloud storage and execution of scripts, creating distributed computing environment. Rich feature set of Rapidminer Studio and various services provided by its developers were the main reason for choosing this system as a development environment for informational-computation resources of the project.

\section{The database supporting metagenomic analyses}
\label{sec:dbsupp}

Assessment of world experience of organization of scientific research in the field of Data Science showed that the use of cloud technologies is a necessary basis for the interaction organization of the researchers. A specialized data storage should be a unit of the environment to ensure effective user access and computing processes to the data of research.

Database for microbiome-based metagenomic analysis data (Fig. 2.) provides the storage facility on all the stages of the microbiome studies from the probe sampling to the publication of the scientific meaningful results. The scheme in the Fig. 2 represents database structure as an ER-diagram. The scheme contains data about sampling, analysis of physicochemical and biological parameters of the probes, the sequencing results, the applied equipment and software, taxonomic databases, methods of the analysis of the collected material, publications of the obtained results and the participated researchers. It also allows us to store the processing scripts of analysis of metagenomic data, including software tools, commands, and configuration files. The latter function allows one to save the state of the computational process and restart it from the specified point.

\begin{figure}[t]\centering
Fig. 2. \caption{A general schema representation of the database for microbiome research based on metagenomic analysis}
\end{figure}

The model is implemented by means of Django framework (https://www.djangoproject.com). The framework supports automatic definition of rational table structures representing many-to-many relations, and generation a customizable interface for the administrative panel, allowing testing the developed model. These same tools are used for the implementation of the project web site.

Cloud-based storage and dedicated storage of the metagenomic data will allow one to create online services for joint processing of sequencing data from different studies by specialized software and to publish information in the Internet. In order to achieve this goal, the following must be carried out:
\begin{itemize}
\item a software interface implementation for data access;
\item filling in the database with information collected and processed as a result of studies of the microbiome of lake Baikal in 2009-2015;
\item realization of the scenario design and execution to support the metagenomic data analysis in a distributed computing environment.
\end{itemize}

\section{Distributive Computing Planning}
\label{sec:distr}

\ldots{} and the parallel execution is provided by the instrumental system ORLANDO.


\subsection{Subsection Heading Here}
Subsection text here.


\subsubsection{Subsubsection Heading Here}
Subsubsection text here.


% An example of a floating figure using the graphicx package.
% Note that \label must occur AFTER (or within) \caption.
% For figures, \caption should occur after the \includegraphics.
% Note that IEEEtran v1.7 and later has special internal code that
% is designed to preserve the operation of \label within \caption
% even when the captionsoff option is in effect. However, because
% of issues like this, it may be the safest practice to put all your
% \label just after \caption rather than within \caption{}.
%
% Reminder: the "draftcls" or "draftclsnofoot", not "draft", class
% option should be used if it is desired that the figures are to be
% displayed while in draft mode.
%
%\begin{figure}[!t]
%\centering
%\includegraphics[width=2.5in]{myfigure}
% where an .eps filename suffix will be assumed under latex,
% and a .pdf suffix will be assumed for pdflatex; or what has been declared
% via \DeclareGraphicsExtensions.
%\caption{Simulation results for the network.}
%\label{fig_sim}
%\end{figure}

% Note that IEEE typically puts floats only at the top, even when this
% results in a large percentage of a column being occupied by floats.


% An example of a double column floating figure using two subfigures.
% (The subfig.sty package must be loaded for this to work.)
% The subfigure \label commands are set within each subfloat command,
% and the \label for the overall figure must come after \caption.
% \hfil is used as a separator to get equal spacing.
% Watch out that the combined width of all the subfigures on a
% line do not exceed the text width or a line break will occur.
%
%\begin{figure*}[!t]
%\centering
%\subfloat[Case I]{\includegraphics[width=2.5in]{box}%
%\label{fig_first_case}}
%\hfil
%\subfloat[Case II]{\includegraphics[width=2.5in]{box}%
%\label{fig_second_case}}
%\caption{Simulation results for the network.}
%\label{fig_sim}
%\end{figure*}
%
% Note that often IEEE papers with subfigures do not employ subfigure
% captions (using the optional argument to \subfloat[]), but instead will
% reference/describe all of them (a), (b), etc., within the main caption.
% Be aware that for subfig.sty to generate the (a), (b), etc., subfigure
% labels, the optional argument to \subfloat must be present. If a
% subcaption is not desired, just leave its contents blank,
% e.g., \subfloat[].


% An example of a floating table. Note that, for IEEE style tables, the
% \caption command should come BEFORE the table and, given that table
% captions serve much like titles, are usually capitalized except for words
% such as a, an, and, as, at, but, by, for, in, nor, of, on, or, the, to
% and up, which are usually not capitalized unless they are the first or
% last word of the caption. Table text will default to \footnotesize as
% IEEE normally uses this smaller font for tables.
% The \label must come after \caption as always.
%
%\begin{table}[!t]
%% increase table row spacing, adjust to taste
%\renewcommand{\arraystretch}{1.3}
% if using array.sty, it might be a good idea to tweak the value of
% \extrarowheight as needed to properly center the text within the cells
%\caption{An Example of a Table}
%\label{table_example}
%\centering
%% Some packages, such as MDW tools, offer better commands for making tables
%% than the plain LaTeX2e tabular which is used here.
%\begin{tabular}{|c||c|}
%\hline
%One & Two\\
%\hline
%Three & Four\\
%\hline
%\end{tabular}
%\end{table}


% Note that the IEEE does not put floats in the very first column
% - or typically anywhere on the first page for that matter. Also,
% in-text middle ("here") positioning is typically not used, but it
% is allowed and encouraged for Computer Society conferences (but
% not Computer Society journals). Most IEEE journals/conferences use
% top floats exclusively.
% Note that, LaTeX2e, unlike IEEE journals/conferences, places
% footnotes above bottom floats. This can be corrected via the
% \fnbelowfloat command of the stfloats package.




\section{Conclusion}
Modern problems of development of a distributed software environment for the implementation of organizational, informational and computational resources for scientific microbiological studies based on metagenome analysis are presented in the article. A generalized domain model of system-level is conducted, as well as the requirements are stated to the development environment and problems to be solved. A computational model of the process of analysis of the amplicons is being constructed and implemented. Aspect of informational supply of the computational process is represented by realization of the problem of cloud storage for computing processes (scenarios of metagenomic data processing), as well as by construction of a database for storing input and intermediate data, results of the scenarios execution. The database is used as a basis of an information portal construction for processing metagenomic data and presenting the results of scientific community.



% conference papers do not normally have an appendix


% use section* for acknowledgment
\section*{Acknowledgment}

The results obtained with the support of the project of Irkutsk scientific center of SB RAS No 4.1.2 and with the use of the network infrastructure of Telecommunication center of collective use ``Integrated information-computational network of Irkutsk scientific-educational complex'' (\url{http://net.icc.ru}).

% trigger a \newpage just before the given reference
% number - used to balance the columns on the last page
% adjust value as needed - may need to be readjusted if
% the document is modified later
%\IEEEtriggeratref{8}
% The "triggered" command can be changed if desired:
%\IEEEtriggercmd{\enlargethispage{-5in}}

% references section

% can use a bibliography generated by BibTeX as a .bbl file
% BibTeX documentation can be easily obtained at:
% http://www.ctan.org/tex-archive/biblio/bibtex/contrib/doc/
% The IEEEtran BibTeX style support page is at:
% http://www.michaelshell.org/tex/ieeetran/bibtex/
%\bibliographystyle{IEEEtran}
% argument is your BibTeX string definitions and bibliography database(s)
%\bibliography{IEEEabrv,../bib/paper}
%
% <OR> manually copy in the resultant .bbl file
% set second argument of \begin to the number of references
% (used to reserve space for the reference number labels box)
\begin{thebibliography}{11}

\bibitem{IEEEhowto:kopka}
H.~Kopka and P.~W. Daly, ``A Guide to \LaTeX,'' 3rd~ed.\hskip 1em plus
  0.5em minus 0.4em\relax Harlow, England: Addison-Wesley, 1999.
\bibitem{Bizer} Ch. Bizer, T. Heath, T. Berners-Lee. ``Linked Data – The Story So Far,'' Semantic Web and Information Systems. 2009. Vol. 5 (3). pp. 1–22.
\bibitem{Cherk} E. Cherkashin, I. Orlova. Instrumental tools for construction of the digital archives of the documents based on Linked Data. Modern technologies, System analysis, Modeling. 4(56) 2017 pp. 100-107 (in Russian) \texttt{DOI:} \url{10.26731/1813-9108.2017.4(56).100-107}, \texttt{URL:} \url{http://stsam.irgups.ru/sites/default/files/articles\_pdf\_files/100-107.pdf}
% \bibitem{Kopay} N. Kopaigorodsky. Use of ontologies in semantic information systems. // Journal "Ontology of Design", 4(14), 2014, p. 78--89 (in Russian) \texttt{URL:} \url{http://agora.guru.ru/scientific\_journal/files/Ontology\_Of\_Designing\_4\_2014\_opt1.pdf}
% \bibitem{MDA} D. Frankel. Model Driven Architecture: Applying MDA to Enterprise Computing. Wiley; 1 edition, 2003, 352 p.
%  \bibitem{Clio} J. Wielemaker, W. Beek, M. Hildebrand, J. Ossenbruggen. ClioPatria: A SWI-Prolog Infrastructure for the Semantic Web. Semantic Web, vol. 7, no. 5, 2016, pp. 529-541. \texttt{DOI:} \url{10.3233/SW-150191}

% \bibitem{org} Org mode fantastic examples: \url{http://ehneilsen.net/notebook/orgExamples/org-examples.html}.
% \bibitem{odmprof} ODM UML profile for OWL: \url{http://www.omg.org/spec/ODM/1.0/PDF/}.
% \bibitem{odnex} OMG Ontology Domain Modeling example: \url{https://thematix.com/tools/vom/}.
% \bibitem{odmvis} OWL UML Visualizer: \url{http://owlgred.lumii.lv/}.
% \bibitem{uml2owd} UMLtoOWL: Converter from UML to OWL: \url{http://www.sfu.ca/~dgasevic/projects/UMLtoOWL/}. Uses XSLT to convert from XMI to OWL.
% \bibitem{atmo3} AToM3: A tool for multi-paradigm modeling. \url{http://atom3.cs.mcgill.ca/}.
% \bibitem{GT} Belghiat, Aissam \& Bourahla, Mustapha. (2012). UML Class Diagrams to OWL Ontologies: A Graph Transformation based Approach. International Journal of Computer Applications. 41. 41-46. 10.5120/5525-7566.
% \bibitem{SWEB} \textbf{Semantic WEB Software Engineering}: \url{http://www.webist.org/Documents/Previous\_Invited\_Speakers/2012/WEBIST2012\_Pan.pdf}. Book: \url{https://www.iospress.nl/book/semantic-web-enabled-software-engineering/}.
\end{thebibliography}




% that's all folks
\end{document}

%%% Local Variables:
%%% End:
